src/unit_tests/wazuh_db/test_wdb_global.c

```diff
@@ -957,6 +957,302 @@ void test_wdb_global_get_agent_max_group_priority_success(void **state)
+    cJSON_AddNumberToObject(j_object, "id", 1);
+    cJSON_AddItemToArray(j_exec_response, j_object);
+
+    will_return(__wrap_wdb_begin2, OS_SUCCESS);
+    will_return(__wrap_cJSON_CreateArray, __real_cJSON_CreateArray());
+    will_return(__wrap_cJSON_CreateArray, __real_cJSON_CreateArray());
+    will_return(__wrap_wdb_stmt_cache, OS_SUCCESS);
+    expect_value(__wrap_sqlite3_bind_int, index, 1);
+    expect_value(__wrap_sqlite3_bind_int, value, last_agent_id);
+    will_return(__wrap_sqlite3_bind_int, SQLITE_OK);
+    will_return(__wrap_wdb_exec_stmt, j_exec_response);
+    expect_function_call(__wrap_cJSON_Delete);
+
+    /* wdb_global_select_group_belong */
+    cJSON *root = cJSON_Parse("[\"default\",\"new_group\"]");
+    will_return(__wrap_wdb_begin2, OS_SUCCESS);
+    will_return(__wrap_wdb_stmt_cache, OS_SUCCESS);
+    expect_value(__wrap_sqlite3_bind_int, index, 1);
+    expect_value(__wrap_sqlite3_bind_int, value, last_agent_id+1);
+    will_return(__wrap_sqlite3_bind_int, SQLITE_OK);
+    will_return(__wrap_wdb_exec_stmt_single_column, root);
+
+    /* wdb_global_set_agent_groups_sync_status */
+    expect_value(__wrap_wdb_init_stmt_in_cache, statement_index, WDB_STMT_GLOBAL_GROUP_SYNC_SET);
+    will_return(__wrap_wdb_init_stmt_in_cache, NULL);
+    expect_string(__wrap__merror, formatted_msg, "Cannot set group_sync_status for agent 1");
+
+    result = wdb_global_sync_agent_groups_get(data->wdb, condition, last_agent_id, set_synced, get_hash, &j_output);
+
+    char *output = cJSON_PrintUnformatted(j_output);
+    assert_string_equal(output, "[{\"data\":[{\"id\":1,\"groups\":[\"default\",\"new_group\"]}]}]");
+    os_free(output);
+    assert_int_equal(result, WDBC_ERROR);
+    __real_cJSON_Delete(j_exec_response);
+    __real_cJSON_Delete(j_output);
+}
+
+void test_wdb_global_sync_agent_groups_get_due_buffer_full(void **state)
+{
+    wdbc_result result = WDBC_OK;
+    test_struct_t *data  = (test_struct_t *)*state;
+    wdb_groups_sync_condition_t condition = 0;
+    int last_agent_id = 0;
+    bool set_synced = false;
+    bool get_hash = true;
+    cJSON *j_output = NULL;
+    cJSON *j_exec_response = __real_cJSON_CreateArray();
+    cJSON *j_object = cJSON_CreateObject();
+    cJSON_AddNumberToObject(j_object, "id", 1);
+    cJSON_AddItemToArray(j_exec_response, j_object);
+
+    will_return(__wrap_wdb_begin2, OS_SUCCESS);
+    will_return(__wrap_cJSON_CreateArray, __real_cJSON_CreateArray());
+    will_return(__wrap_cJSON_CreateArray, __real_cJSON_CreateArray());
+    will_return(__wrap_wdb_stmt_cache, OS_SUCCESS);
+    expect_value(__wrap_sqlite3_bind_int, index, 1);
+    expect_value(__wrap_sqlite3_bind_int, value, last_agent_id);
+    will_return(__wrap_sqlite3_bind_int, SQLITE_OK);
+    will_return(__wrap_wdb_exec_stmt, j_exec_response);
+    expect_function_call(__wrap_cJSON_Delete);
+
+    /* wdb_global_select_group_belong */
+    cJSON *j_groups = __real_cJSON_CreateArray();
+    for (int i = 0; i < 5000; ++i) {
```

Comment:
why using 5000 instead of  WDB_MAX_RESPONSE_SIZE ?

Reply:
`WDB_MAX_RESPONSE_SIZE` is close to 65536. Using that will slow down the test case unnecessarily. 
 What I forced here is to create a JSON object with 5000 group names of 10 characters long ("test_group").  
 
With that, the condition `if (response_size+agent_len+1 < WDB_MAX_RESPONSE_SIZE) {}` in wdb_global_sync_agent_groups_get() will be evaluated as false and it will return the WDB_DUE response I need.